### ISSUE TITLE
Add optional flag to chpldoc that generates intents for arguments

### ISSUE THREAD

**Issue by benharsh** [chapel-lang/chapel#5739]
While we generally expect users to know the default intent for Chapel types, it could be useful to optionally generate the intents for a chpldoc'd module. I think this is even more useful when considering the new 'ref-if-modified' intent for arrays. This would likely require invoking full resolution, which is why I suggest that such a flag be optional (at least for now).

If the argument is fully generic, I'd be fine if we continued to not generate an intent.

Another potential option: support this with a pragma/attribute on a per-function basis?

**Comment by benharsh** [comment]
@lydia-duncan : for your consideration.

Should we be collecting these in some kind of meta-wishlist for chpldoc?

**Comment by lydia-duncan** [comment]
@benharsh: This is an interesting idea and I think worth considering.

Regarding your meta question, I've been marking chpldoc changes with the area: Tools label.  I think many of the requests could be done separately, so it seems useful to have an issue for each.  My inclination is to add a chpldoc label rather than an issue that tracks all related issues, since the latter requires more active maintenance, but @ben-albrecht seemed disinclined when I first brought it up.  What are your thoughts?

**Comment by benharsh** [comment]
I think the ``Tools`` label makes sense. I guess I'm similarly disinclined to add a ``chpldoc`` label because I'd prefer to keep the number of labels fairly low. Maybe ``Tools`` is good enough for now? I agree that a meta-issue would be additional overhead. Perhaps we can revisit as the number of chpldoc issues increases.

**Comment by ben-albrecht** [comment]
Yeah, searching by the `Tools` label  sufficient for finding all the chpldoc issues today, and adding chpldoc to the search string would easily distinguish chpldoc from chplvis, if we end up with a lot of chplvis issues.
